klippa/src/glyf_loca.rs

```diff
@@ -0,0 +1,304 @@
+//! impl subset() for glyf and loca
+use crate::{estimate_subset_table_size, Plan, SubsetError, SubsetError::SubsetTableError};
+use write_fonts::{
+    read::{
+        tables::{
+            glyf::{
+                CompositeGlyph, CompositeGlyphFlags, Glyf,
+                Glyph::{self, Composite, Simple},
+                SimpleGlyph, SimpleGlyphFlags,
+            },
+            head::Head,
+            loca::Loca,
+        },
+        types::GlyphId,
+        FontRef, TableProvider, TopLevelTable,
+    },
+    FontBuilder,
+};
+
+pub fn subset_glyf_loca(
+    plan: &Plan,
+    font: &FontRef,
+    builder: &mut FontBuilder,
+) -> Result<(), SubsetError> {
+    let Ok(glyf) = font.glyf() else {
+        return Err(SubsetTableError(Glyf::TAG));
+    };
+
+    let Ok(loca) = font.loca(None) else {
+        return Err(SubsetTableError(Loca::TAG));
+    };
+
+    let Ok(head) = font.head() else {
+        return Err(SubsetTableError(Head::TAG));
+    };
+
+    let num_output_glyphs = plan.num_output_glyphs;
+    let mut subset_glyphs = Vec::with_capacity(num_output_glyphs);
+    let mut max_offset: u32 = 0;
+
+    //TODO: support not_def_outline and drop_hints
+    for (_new_gid, old_gid) in &plan.new_to_old_gid_list {
+        match loca.get_glyf(*old_gid, &glyf) {
+            Ok(g) => {
+                let Some(glyph) = g else {
+                    subset_glyphs.push(Vec::new());
+                    continue;
+                };
+                let subset_glyph = subset_glyph(&glyph, plan);
+                let trimmed_len = subset_glyph.len();
+                max_offset += padded_size(trimmed_len) as u32;
+                subset_glyphs.push(subset_glyph);
+            }
+            _ => {
+                return Err(SubsetTableError(Glyf::TAG));
```

Comment:
I suspect at some point you are going to be frustrated by how general these errors are. In this case I would be tempted to have a `MissingGlyph(GylphId)`  error, or something?